### PR TITLE
Added option to disable or perform less MCMC refitting during similar-segment merging.

### LINKED ISSUE #499: POS -> POSITION change in pulldown column name breaks plotting.
This change was made in #444.  Also breaks evaluation scripts.  @LeeTL1220, is backwards compatibility with previous pulldowns desired?  See also #451.

### PR DESCRIPTION
**Pull request by samuelklee**
Exposes more control over MCMC refitting performed during similar-segment merging, which can be useful for development.

Similar-segment merging is currently performed by traversing the genome and merging adjacent segments that have credible intervals that overlap in CR and MAF by a specified amount.  The credible intervals of a merged segment are updated in a heuristic manner.  After traversal, an MCMC fit is performed.  This procedure is repeated until convergence.

Because each MCMC fit is expensive, this yields longer runtimes for samples that are initially very segmented (e.g., LUAD).  For development purposes, it is nice to have more control over this runtime (beyond just limiting the number of iterations of merging allowed, which is already possible).  With the new argument numIterationsSimSegPerFit, one can choose to only refit intermittently.  Evaluation shows that although less segmentation is smoothed out when a refit is not performed after every merge iteration, HAPSEG concordance is not drastically reduced.

This feature should still be useful even if we update the method of similar-segment merging in the future (hopefully to something less heuristic), since I expect it should remain a relatively expensive step.

Also added some basic validation of CLI arguments.

@achevali, note that this also closes #499.

### CODE REVIEW

**Comment by samuelklee** [comment]
@mbabadi, mind taking a look at this when you get a chance?

File: src/main/resources/org/broadinstitute/hellbender/utils/plotter/CNV_plotting_library.R [hunk 1]
```diff
@@ -57,8 +57,8 @@ PlotAlleleFraction = function(snp_df, segments_df){
      contig = segments_df[s,"Chromosome"]
      segment_start = segments_df[s, "Start"]
      segment_end = segments_df[s, "End"]
-     snp_indices=which(snp_df[,"CONTIG"] == contig & snp_df[,"POS"] > segment_start & snp_df[,"POS"] < segment_end)
-     genomic_coordinates = chromosome_starts[contig] + snp_df[ snp_indices,"POS"]
+     snp_indices=which(snp_df[,"CONTIG"] == contig & snp_df[,"POSITION"] > segment_start & snp_df[,"POSITION"] < segment_end)
```
Comment (mbabadi) [inline src/main/resources/org/broadinstitute/hellbender/utils/plotter/CNV_plotting_library.R]: This is already in master.

File: src/test/java/org/broadinstitute/hellbender/tools/exome/ACNVModellerUnitTest.java [hunk 2]
```diff
@@ -119,4 +125,49 @@ public void testMergeSimilarSegments() {
...
+        Assert.assertTrue(!modeller.isModelFit());
+
+        //write final model fit to file; this should force model refit
+        final File finalModeledSegmentsFile =
+                new File(tempDirFile.getAbsolutePath() + "/test-" + FINAL_SEG_FILE_TAG + ".seg");
+        modeller.writeACNVModeledSegmentFile(finalModeledSegmentsFile);
+        Assert.assertTrue(modeller.isModelFit());
+
+        //check final number of segments = 31 (ground truth has 30 segments, so we miss one merge by not performing intermediate refits)
+        final int expectedNumberOfSegments = 31;
```
Comment (mbabadi) [inline src/test/java/org/broadinstitute/hellbender/tools/exome/ACNVModellerUnitTest.java]: Magic number and depends on the input data. Please assign a constant to it in the preamble or better yet, make a truth segmented model file and test equality against it (similar to what you do in `testMergeSimilarSegments`). Also, please consider renaming `testMergeSimilarSegments` to `testMergeSimilarSegmentsWithRefitting`.

File: src/test/java/org/broadinstitute/hellbender/tools/exome/AllelicCNVIntegrationTest.java [hunk 3]
```diff
@@ -63,6 +101,64 @@ public void testACNVWithAllelicPON() {
...
+                "--verbosity", "INFO",
+        };
+        testACNV(arguments, outputPrefix);
+
+        //check that some similar segments were merged
+        final File initialSimilarSegmentsFile = new File(outputPrefix + "-" + AllelicCNV.INITIAL_SEG_FILE_TAG + ".seg");
+        final File finalSimilarSegmentsFile = new File(outputPrefix + "-" + AllelicCNV.FINAL_SEG_FILE_TAG + ".seg");
+        final List<ACNVModeledSegment> initialACNVModeledSegments = SegmentUtils.readACNVModeledSegmentFile(initialSimilarSegmentsFile);
+        final List<ACNVModeledSegment> finalACNVModeledSegments = SegmentUtils.readACNVModeledSegmentFile(finalSimilarSegmentsFile);
+        Assert.assertTrue(initialACNVModeledSegments.size() > finalACNVModeledSegments.size());
```
Comment (mbabadi) [inline src/test/java/org/broadinstitute/hellbender/tools/exome/AllelicCNVIntegrationTest.java]: Please write a warning note in the header of `AllelicCNVIntegrationTest` mentioning that the behavior of the tests depend on the _content_ of the input files and changing them may break the tests.

**Comment by mbabadi** [comment]
Back to you @samuelklee. The comments are mere suggestions. Feel free to rebase and merge.

File: src/test/java/org/broadinstitute/hellbender/tools/exome/ACNVModellerUnitTest.java (hunk 2 above)
Comment (samuelklee) [inline src/test/java/org/broadinstitute/hellbender/tools/exome/ACNVModellerUnitTest.java]: Good catch.  Now that I think about it, I'm not sure what is actually a satisfying test here.  Maybe just require that merging without refitting gets to within a certain threshold of the expected number of segments that merging with refitting finds?  In any case, I'll add comments and get rid of the magic number.

Comment (mbabadi) [inline src/test/java/org/broadinstitute/hellbender/tools/exome/ACNVModellerUnitTest.java]: Well it's difficult to devise a bullet proof test here -- specially if we happen to change some details of the segment merging algorithm in the future (the behavior of the full algorithm with and without refitting is difficult to predict). I think it should be sufficient to warn about it in the header doc :)

**Pull request merged**